src/components/Orcid.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Fragment } from "react";
+import EduIDButton from "components/EduIDButton";
+import { FormattedMessage } from "react-intl";
+import { PDOrcid } from "apis/eduidPersonalData";
+
+const orcidIcon = require("../../img/vector_iD_icon-w.svg");
+
+export interface OrcidProps {
+  orcid: PDOrcid;
+  handleOrcidConnect: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  handleOrcidDelete: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+function Orcid(props: OrcidProps): JSX.Element {
+  if (!props.orcid || !Object.keys(props.orcid)?.length) {
```

Comment:
It seems like instead of checking if the object contains any keys here, we could check for the property of the object we actually will use below instead.

`if (! props.orcid?.id) {`